Update to diesel 2.0

When changing the version to 2.0, the build fails with

```rust
> rustup run stable cargo build
    Updating crates.io index
   Compiling version_check v0.9.4
   Compiling syn v1.0.107
   Compiling serde v1.0.152
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
   Compiling serde_derive v1.0.152
   Compiling diesel_derives v2.0.1
   Compiling serde_json v1.0.91
   Compiling diesel v2.0.3
   Compiling diesel_json v0.1.1 (/home/jayvdb/rust/diesel_json)
warning: #[sql_type] attribute form is deprecated
  = help: use `#[diesel(sql_type = sql_types::Jsonb)]` instead

warning: #[sql_type] attribute form is deprecated
  = help: use `#[diesel(sql_type = sql_types::Jsonb)]` instead

error[E0107]: this struct takes 1 generic argument but 2 generic arguments were supplied
  --> src/lib.rs:84:38
   |
84 |         out: &mut diesel::serialize::Output<W, Pg>,
   |                                      ^^^^^^    -- help: remove this generic argument
   |                                      |
   |                                      expected 1 generic argument
   |
note: struct defined here, with 1 generic parameter: `DB`
  --> /home/jayvdb/.cargo/registry/src/github.com-1ecc6299db9ec823/diesel-2.0.3/src/serialize.rs:35:12
   |
35 | pub struct Output<'a, 'b, DB>
   |            ^^^^^^         --

For more information about this error, try `rustc --explain E0107`.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,9 +16,9 @@ categories = ["database", "encoding"]
 [dependencies]
 serde = { version = "1.0", default-features = false, features = ["derive"] }
 serde_json = { version = ">=0.8.0, <2.0" }
-diesel = { version = "1.4", features = ["postgres", "serde_json"] }
+diesel = { version = "2.0", features = ["postgres", "serde_json"] }
 
 
 [dev-dependencies]
-diesel_migrations = "1.4.0"
+diesel_migrations = "2.0"
 dotenv = "0.15.0"

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -5,7 +5,7 @@ extern crate diesel;
 #[macro_use]
 extern crate serde;
 
-use diesel::pg::Pg;
+use diesel::pg::{Pg, PgValue};
 use diesel::sql_types;
 use diesel::{deserialize::FromSql, serialize::ToSql};
 use serde::de::DeserializeOwned;
@@ -30,7 +30,7 @@ use std::ops::{Deref, DerefMut};
 /// ```
 #[derive(FromSqlRow, AsExpression, Serialize, Deserialize, Debug, Clone)]
 #[serde(transparent)]
-#[sql_type = "sql_types::Jsonb"]
+#[diesel(sql_type = sql_types::Jsonb)]
 pub struct Json<T: Sized>(pub T);
 
 impl<T> Json<T> {
@@ -69,7 +69,7 @@ impl<T> FromSql<sql_types::Jsonb, Pg> for Json<T>
 where
     T: std::fmt::Debug + DeserializeOwned,
 {
-    fn from_sql(bytes: Option<&[u8]>) -> diesel::deserialize::Result<Self> {
+    fn from_sql(bytes: PgValue) -> diesel::deserialize::Result<Self> {
         let value = <serde_json::Value as FromSql<sql_types::Jsonb, Pg>>::from_sql(bytes)?;
         Ok(Json(serde_json::from_value::<T>(value)?))
     }
@@ -79,12 +79,9 @@ impl<T> ToSql<sql_types::Jsonb, Pg> for Json<T>
 where
     T: std::fmt::Debug + Serialize,
 {
-    fn to_sql<W: std::io::Write>(
-        &self,
-        out: &mut diesel::serialize::Output<W, Pg>,
-    ) -> diesel::serialize::Result {
+    fn to_sql(&self, out: &mut diesel::serialize::Output<Pg>) -> diesel::serialize::Result {
         let value = serde_json::to_value(self)?;
-        <serde_json::Value as ToSql<sql_types::Jsonb, Pg>>::to_sql(&value, out)
+        <serde_json::Value as ToSql<sql_types::Jsonb, Pg>>::to_sql(&value, &mut out.reborrow())
     }
 }
 

--- a/tests/postgresql.rs
+++ b/tests/postgresql.rs
@@ -37,7 +37,7 @@ struct JsonbTest {
 use schema::jsonb_test;
 
 #[derive(Insertable)]
-#[table_name = "jsonb_test"]
+#[diesel(table_name = jsonb_test)]
 struct JsonbTestInsert {
     nullable: Option<Json<ComplexData>>,
     not_nullable: Json<ComplexData>,
@@ -67,10 +67,10 @@ fn test_insert_and_select() {
         },
     ];
 
-    let connection = setup::transaction_connection();
+    let mut connection = setup::transaction_connection();
     let result = diesel::insert_into(schema::jsonb_test::dsl::jsonb_test)
         .values(&test_data)
-        .get_results::<JsonbTest>(&connection)
+        .get_results::<JsonbTest>(&mut connection)
         .expect("Error inserting and retrieving inserted data");
     assert_eq!(result.len(), 2);
 

--- a/tests/setup/mod.rs
+++ b/tests/setup/mod.rs
@@ -2,6 +2,7 @@ use std::sync::Once;
 
 use diesel::pg::PgConnection;
 use diesel::prelude::Connection;
+use diesel_migrations::{FileBasedMigrations, MigrationHarness};
 
 static INIT: Once = Once::new();
 
@@ -16,20 +17,19 @@ fn establish_connection() -> PgConnection {
 /// Setup function that is only run once, even if called multiple times.
 pub fn setup() {
     INIT.call_once(|| {
-        let migration_dir = diesel_migrations::find_migrations_directory().unwrap();
-        diesel_migrations::run_pending_migrations_in_directory(
-            &establish_connection(),
-            &migration_dir,
-            &mut std::io::sink(),
-        )
-        .expect("Migrations didn't succeed");
+        let migrations = FileBasedMigrations::find_migrations_directory().unwrap();
+        let mut connection = establish_connection();
+        connection
+            .run_pending_migrations(migrations)
+            .expect("Migrations didn't succeed");
+        connection.begin_test_transaction().unwrap();
     });
 }
 
 pub fn transaction_connection() -> PgConnection {
-    let result = establish_connection();
+    let mut connection = establish_connection();
     // #[cfg(feature = "sqlite")]
-    // result.execute("PRAGMA foreign_keys = ON").unwrap();
-    result.begin_test_transaction().unwrap();
-    result
+    // connection.execute("PRAGMA foreign_keys = ON").unwrap();
+    connection.begin_test_transaction().unwrap();
+    connection
 }

--- a/tests/tests.rs
+++ b/tests/tests.rs
@@ -1,4 +1,3 @@
-#[macro_use]
 extern crate diesel;
 
 pub mod setup;
@@ -13,11 +12,13 @@ fn test_diesel_connection() {
 
     #[derive(QueryableByName, Debug, PartialEq)]
     struct SimpleQuery {
-        #[sql_type = "diesel::sql_types::Integer"]
+        #[diesel(sql_type = diesel::sql_types::Integer)]
         number: i32,
     }
 
-    let connection = setup::transaction_connection();
-    let result: Vec<SimpleQuery> = sql_query("SELECT 1 as number").load(&connection).unwrap();
+    let mut connection = setup::transaction_connection();
+    let result: Vec<SimpleQuery> = sql_query("SELECT 1 as number")
+        .load(&mut connection)
+        .unwrap();
     assert_eq!(result, vec![SimpleQuery { number: 1 }]);
 }

